oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/web/DefaultOAuth2AuthorizationRequestResolver.java

```diff
@@ -52,6 +57,7 @@
 	private final ClientRegistrationRepository clientRegistrationRepository;
 	private final AntPathRequestMatcher authorizationRequestMatcher;
 	private final StringKeyGenerator stateGenerator = new Base64StringKeyGenerator(Base64.getUrlEncoder());
+	private final StringKeyGenerator codeVerifierGenerator = new Base64StringKeyGenerator(Base64.getUrlEncoder().withoutPadding());
```

Comment:
@jgrandja we should use `withoutPadding()` to ensure that we generate a code_verifier that doesn't contain any `=` padding characters at the end of the String since the spec only allows `[A-Z] / [a-z] / [0-9] / "-" / "." / "_" / "~"`.

As for the default length, I'll set the key length to 96 (bytes) rather than the default 32 (bytes) because it translates to the maximum base64 url-encoded string length of 128.

Reply:
> ensure that we generate a code_verifier that doesn't contain any `=` padding characters

Gotchya, makes sense.

> As for the default length, I'll set the key length to 96 (bytes)

Great